drivers/sensor/sensor_sim/sensor_sim.c

```diff
@@ -211,9 +214,12 @@ static double generate_pseudo_random(void)
  */
 static double generate_sine(double offset, double amplitude)
 {
-	uint32_t time = k_uptime_get_32();
+	static const uint32_t period_ms = 10000;
 
-	return offset + amplitude * sin(time % 65535);
+	double time = (k_uptime_get_32() % period_ms);
```

Comment:
once the cast is gone there is also no need to brackets

Reply:
done